tests/test_upload_client.py

```diff
@@ -1,20 +1,395 @@
-from unittest import TestCase
+import json
+import os
+import re
+from urllib.parse import parse_qs
+from xml.etree.ElementTree import Element
 
+import boto3
+import httmock
+from lxml import etree
+import pytest
+from xmltodict import parse
+
+from gdc_client.upload import GDCUploadClient
 from gdc_client.upload.client import create_resume_path
 
 
-class UploadClientTest(TestCase):
-    def setup(self):
-        pass
-
-    def test_create_resume_path(self):
-        # don't need to test if there's no file given
-        # that is checked in multipart_upload()
-        tests = ["/path/to/file.yml", "path/to/file.yml", "file.yml"]
-        results = [
-            "/path/to/resume_file.yml",
-            "path/to/resume_file.yml",
-            "resume_file.yml",
-        ]
-        for i, t in enumerate(tests):
-            assert create_resume_path(t) == results[i]
+# This is based entirely on the known GraphQL query that is sent to backend
+GRAPHQL_RE = (
+    "^query \w+ \{ "
+    "(?P<type>\w+)"
+    " \(id: \""
+    "(?P<node_id>[a-z0-9-]+)"
+    "\"\) \{ "
+    "(?P<fields>[\w ]+)"
+    " } }$"
+)
+FIVE_MB = 5 * 1024 * 1024
+
+
+@pytest.fixture
+def mock_dir(tmpdir):
+    """Create directory for upload files"""
+    bam_dir = tmpdir.mkdir("bams")
+    return bam_dir
+
+
+@pytest.fixture
+def mock_dir_path(mock_dir):
+    return str(mock_dir)
+
+
+@pytest.fixture
+def mock_files(mock_dir):
+    """Create 2 files: 10MB and 15MB in size"""
+    file_name1 = "fake_bam_1.bam"
+    file_name2 = "fake_bam_2.bam"
+
+    # Create 10MB file
+    f1 = mock_dir.join(file_name1)
+    size1 = 10 * 1024 * 1024
+    f1.write("a" * size1)
+
+    # Create 15MB file
+    f2 = mock_dir.join(file_name2)
+    size2 = 15 * 1024 * 1024
+    f2.write("b" * size2)
+
+    return file_name1, file_name2
+
+
+@pytest.fixture
+def mock_graphql_responses(mock_files):
+    file_id1 = "file-id-1"
+    file_id2 = "file-id-2"
+
+    metadata = {
+        file_id1: {
+            "type": "submitted_unaligned_reads",
+            "project_id": "GDC-MISC",
+            "file_name": mock_files[0],
+        },
+        file_id2: {
+            "type": "submitted_unaligned_reads",
+            "project_id": "GDC-MISC",
+            "file_name": mock_files[1],
+        },
+    }
+
+    @httmock.urlmatch(netloc="localhost", method="POST", path="/v0/submission/graphql")
```

Comment:
You weren't kidding when you said there was a lot of mocking to do.  Thanks for adding these to allow us to increase our test coverage.

For the long-term, we should think about how we want to perform integration tests for any of our code that depends on external services.  Mocking services like this fragile and problematic to maintain in the long-term.  
(not something we need to figure out for this pr though)

Reply:
with jenkins we should just make a pipeline for `gdc-client` specifically and talk to the deployed services. we could potentially mock more aggressively here, e.g. don't even have a `moto` or `httmock` and just mock `requests`, but meh

Reply:
I like the Jenkins idea.  I'll add it to my list of potential solutions for this testing problem.